### PR TITLE
Compiling area cleaned/updated

### PR DESCRIPTION
**Pull request by SaturnusDJ**
- Moved icons dir into build dir, updated references to it
- Moved dmgbg.png into build dir, updated references to it
- Moved dmg.json into build dir, updated references/metadata in it and to it
- Changed branding in build.json
- Added buildDir in build.json so we have control over the output
  location
- Added cacheDir in build.json so cache is in a better location and not
  in root anymore
- Updated the paths to the changes above
- The first line of installer-script.iss opened with strange signs,
  fixed
- Changed branding and metadata in installer-script.iss
- Updated installer-script.iss with new paths
- Made paths in installer-script.iss relative, so it works for everybody
  using the repo
- Updated gitignore
- Confirmed the compiling proces is still working (None related bug:
  https://github.com/tinderjs/tinder-desktop/issues/5#issuecomment-207819152)

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -36,7 +39,7 @@ function createNW() {
   var archiver = require('archiver');
   var archive = archiver('zip');
 
-  var output = fs.createWriteStream('./build/' + appName + '/tinder-' + appPkg.version + '.nw');
+  var output = fs.createWriteStream(buildDir + '/tinder-' + appPkg.version + '.nw');
```
Comment (stuwil) [inline build.js]: Can we change this to tinder-desktop-{{version}}.nw?

File: build.js [hunk 2]
```diff
@@ -3,16 +3,19 @@ console.log('Before running, make sure versions are updated in both package.json
 
 var NwBuilder = require('node-webkit-builder');
 var appPkg = require('./desktop-app/package.json');
-var appName = 'Tinder⁺⁺';
+var appName = 'Tinder Desktop';
```
Comment (SaturnusDJ) [inline build.js]: `var appName = 'Tinder Desktop';` is probably internal too so can be tinder-desktop, but then the paths in installer-script.iss containing 'Tinder Desktop' need to be renamed accordingly.

Comment (stuwil) [inline build.js]: Fine with that... we really shouldn't have spaces in our build path.

File: build.js (hunk 1 above)
Comment (SaturnusDJ) [inline build.js]: Done in e02752d

File: build/installer-script.iss [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 ; SEE THE DOCUMENTATION FOR DETAILS ON CREATING INNO SETUP SCRIPT FILES!
 
-#define MyAppName "Tinder++"
+#define MyAppName "Tinder Desktop"
 #define MyAppVersion "1.2.1"
-#define MyAppPublisher "VibraMedia, LLC"
-#define MyAppURL "http://www.tinderplusplus.com/"
-#define MyAppExeName "Tinder⁺⁺.exe"
+#define MyAppPublisher "TinderJS, Github Organization"
+#define MyAppURL "https://tinderjs.com"
```
Comment (stuwil) [inline build/installer-script.iss]: Use https://github.com/tinderjs/tinder-desktop

**Pull request merged**